rq/job.py

```diff
@@ -78,7 +91,7 @@ class Job:
     @classmethod
     def create(cls, func, args=None, kwargs=None, connection=None,
                result_ttl=None, ttl=None, status=None, description=None,
-               depends_on=None, timeout=None, id=None, origin=None, meta=None,
+               depends_on: Dependency = None, timeout=None, id=None, origin=None, meta=None,
```

Comment:
`depends_on` needs to be backwards compatible so we don't break existing codes. We'll need to check whether this is a `Dependency` instance before we use the new code path.